src/GUI/UI/Books/TownPortalBook.cpp

```diff
@@ -90,7 +139,11 @@ void GUIWindow_TownPortalBook::Update() {
     townPortalWindow.uFrameZ = game_viewport_z;
     townPortalWindow.uFrameW = game_viewport_w;
 
-    for (int i = 0; i < TOWN_PORTAL_DESTINATION_COUNT; ++i) {
+    int count = TOWN_PORTAL_DESTINATION_COUNT;
+    if (engine->config->debug.TownPortal.value()) {
+        count = TOWN_PORTAL_DESTINATION_COUNT_WITH_CHEATS;
+    }
+    for (int i = 0; i < count; ++i) {
         if (pParty->_questBits[townPortalList[i].qBit] || engine->config->debug.TownPortal.value()) {
             render->ZDrawTextureAlpha(townPortalButtonsPos[i].x / 640.0f, townPortalButtonsPos[i].y / 480.0f, ui_book_townportal_icons[i], i + 1);
```

Comment:
We have a DrawMasked render function - you could draw the new locations greyed out a bit here